mp/src/public/vgui_controls/ProgressBar.h

```diff
@@ -20,87 +20,108 @@ namespace vgui
-	int GetDrawnSegmentCount();
+    virtual void ApplySettings(KeyValues *inResourceData);
+    virtual void GetSettings(KeyValues *outResourceData);
+    void InitSettings() override;
 
-	enum ProgressDir_e
-	{
-		PROGRESS_EAST,
-		PROGRESS_WEST,
-		PROGRESS_NORTH,
-		PROGRESS_SOUTH
-	};
+    // returns the number of segment blocks drawn
+    int GetDrawnSegmentCount();
 
-	int GetProgressDirection() const { return m_iProgressDirection; }
-	void SetProgressDirection( int val ) { m_iProgressDirection = val; }
+    enum ProgressDir_e
+    {
+        PROGRESS_EAST,
+        PROGRESS_WEST,
+        PROGRESS_NORTH,
+        PROGRESS_SOUTH
+    };
+
+    int GetProgressDirection() const { return m_iProgressDirection; }
+    void SetProgressDirection(int val) { m_iProgressDirection = val; }
+
+    void SetSubdivMarksVisible(bool bVisible) { m_bSubdivMarksVisible = bVisible; }
+    void SetSubdivMarksColor(Color color) { m_SubdivMarksColor = color; }
+    void SetSubdivMarksCount(int count) { m_iSubdivMarksCount = count; }
+    void SetSubdivMarksWidth(int width) { m_iSubdivMarksWidth = width; }
+    void SetSubdivMarksHidden(bool bHidden) { m_bSubdivMarksHidden = bHidden; }
+
+    bool  GetSubdivMarksVisible() { return m_bSubdivMarksVisible; }
+    Color GetSubdivMarksColor() { return m_SubdivMarksColor; }
+    int   GetSubdivMarksCount() { return m_iSubdivMarksCount; }
+    int   GetSubdivMarksWidth() { return m_iSubdivMarksWidth; }
+    bool  GetSubdivMarksHidden() { return m_bSubdivMarksHidden; }
 
 protected:
-	virtual void Paint();
-    void PerformLayout() OVERRIDE;
-	void PaintSegment( int &x, int &y, int tall, int wide );
-	virtual void PaintBackground();
-	virtual void ApplySchemeSettings(IScheme *pScheme);
-	MESSAGE_FUNC_PARAMS( OnDialogVariablesChanged, "DialogVariables", dialogVariables );
-	/* CUSTOM MESSAGE HANDLING
-		"SetProgress"
-			input:	"progress"	- float value of the progress to set
-	*/
+    virtual void Paint();
+    void PerformLayout() override;
+    void PaintSegment(int &x, int &y, int tall, int wide);
+    void PaintSubdivMarks(int x, int y, int tall, int wide);
+    virtual void PaintBackground();
+    virtual void ApplySchemeSettings(IScheme *pScheme);
+    MESSAGE_FUNC_PARAMS(OnDialogVariablesChanged, "DialogVariables", dialogVariables);
+    /* CUSTOM MESSAGE HANDLING
+        "SetProgress"
+            input:	"progress"	- float value of the progress to set
+    */
 
 protected:
-	int m_iProgressDirection;
-	float _progress;
+    int m_iProgressDirection;
+    float _progress;
 
 private:
-	int   _segmentCount;
-	int _segmentGap;
-	int _segmentWide;
-	int m_iBarInset;
-	int m_iBarMargin;
-	CUtlString m_pszDialogVar;
+    int _segmentCount;
+    int _segmentGap;
+    int _segmentWide;
+    int m_iBarInset;
+    int m_iBarMargin;
+    CUtlString m_pszDialogVar;
     Label *m_pProgressPercent;
+
+    bool m_bSubdivMarksVisible;
+    Color m_SubdivMarksColor;
+    int m_iSubdivMarksCount;
+    int m_iSubdivMarksWidth;
```

Comment:
To be honest, these should probably be `CPanelAnimationVar` types. For example, for the Color one it would allow you to use a scheme defined color instead of having to hardcode the color RGBA every time. It would also handle loading the settings for you so you don't have to do it in LoadSettings.

Reply:
Interesting I never thought you could have int animation vars, partly because of an old comment in the speedometer saying that they have to be floats. I guess they need to be floats if they have an animation?

Anyways I converted them all to `CPanelAnimationVar` except the visible (the old 'hidden' field).

Reply:
Yeah they have to be floats if you're actually animating it, because the animation controller can't handle linearly interpolating between ints, but can for floats which are then casted to ints. It's weird and something to probably look into more but we don't have many animations.